### PR TITLE
Feature/add soda spark task

### PR DESCRIPTION
**Pull request by AlessandroLollo**
## Summary
This PR adds the ability to run Soda scans to Spark DataFrames using soda-spark library

## Importance
Users can now run Data Quality checks on Spark DataFrames.

This PR:

- [x] adds new tests (if appropriate)
- [x] adds a change file in the `changes/` directory (if appropriate)
- [x] updates docstrings for any new functions or function arguments, including `docs/outline.toml` for API reference docs (if appropriate)

### CODE REVIEW

**Comment by AlessandroLollo** [comment]
The issue is
```
Building wheel for sasl (setup.py) ... - \ error
  ERROR: Command errored out with exit status 1:
   command: /usr/local/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-yq32rmnm/sasl/setup.py'"'"'; __file__='"'"'/tmp/pip-install-yq32rmnm/sasl/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-r228tua8
       cwd: /tmp/pip-install-yq32rmnm/sasl/
  Complete output (30 lines):
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.linux-x86_64-3.8
  creating build/lib.linux-x86_64-3.8/sasl
  copying sasl/__init__.py -> build/lib.linux-x86_64-3.8/sasl
  running egg_info
  writing sasl.egg-info/PKG-INFO
  writing dependency_links to sasl.egg-info/dependency_links.txt
  writing requirements to sasl.egg-info/requires.txt
  writing top-level names to sasl.egg-info/top_level.txt
  reading manifest file 'sasl.egg-info/SOURCES.txt'
  reading manifest template 'MANIFEST.in'
  adding license file 'LICENSE.txt'
  writing manifest file 'sasl.egg-info/SOURCES.txt'
  copying sasl/saslwrapper.cpp -> build/lib.linux-x86_64-3.8/sasl
  copying sasl/saslwrapper.h -> build/lib.linux-x86_64-3.8/sasl
  copying sasl/saslwrapper.pyx -> build/lib.linux-x86_64-3.8/sasl
  running build_ext
  building 'sasl.saslwrapper' extension
  creating build/temp.linux-x86_64-3.8
  creating build/temp.linux-x86_64-3.8/sasl
  gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -Isasl -I/usr/local/include/python3.8 -c sasl/saslwrapper.cpp -o build/temp.linux-x86_64-3.8/sasl/saslwrapper.o
  In file included from sasl/saslwrapper.cpp:629:
  sasl/saslwrapper.h:22:10: fatal error: sasl/sasl.h: No such file or directory
     22 | #include <sasl/sasl.h>
        |          ^~~~~~~~~~~~~
  compilation terminated.
  error: command 'gcc' failed with exit status 1
  ----------------------------------------
  ERROR: Failed building wheel for sasl
  Running setup.py clean for sasl
  Building wheel for pure-sasl (setup.py) ... - done
```

I'm not sure how to fix it though.

**Comment by AlessandroLollo** [comment]
Tests are failing because of something related to `soda-sql` and `soda-spark`.
I will investigate.

File: setup.py [hunk 1]
```diff
@@ -91,6 +91,7 @@ def run(self):
     "dremio": ["pyarrow >= 5.0.0"],
     "exasol": ["pyexasol >= 0.16.1"],
     "sodasql": ["soda-sql >= 2.0.0b25"],
```
Comment (vijaykiran) [inline setup.py:93]: This should not be added .. the soda-sql deps are transitive from soda-spark

Comment (AlessandroLollo) [reply setup.py]: Hey @vijaykiran 😊 
Thanks for the suggestion!
I tried the change, but still getting the same error https://app.circleci.com/pipelines/github/PrefectHQ/prefect/10298/workflows/1a2da562-1dbf-40be-bd14-e7573bbb3f42/jobs/136120

File: setup.py [hunk 2]
```diff
@@ -90,7 +90,7 @@ def run(self):
     "twitter": ["tweepy >= 3.5, < 4.0"],
     "dremio": ["pyarrow >= 5.0.0"],
     "exasol": ["pyexasol >= 0.16.1"],
-    "sodasql": ["soda-sql >= 2.0.0b25"],
+    "soda": ["soda-spark >= 0.2.1"],
```
Comment (madkinsz) [inline setup.py:93]: Removing the existing extra is a breaking change. You'll need to retain that. We can drop it in 1.0, although I'm curious why it's not needed anymore. `soda-spark` requires `soda-sql` I presume?

File: src/prefect/tasks/sodaspark/sodaspark_tasks.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+        Args:
+            - scan_def (str, optional): scan definition.
+              Can be either a path to a YAML file containing the scan definition
+              or the scan definition given as a valid YAML string
+            - df (pyspark.sql.DataFrame, optional): Spark DataFrame.
+              DataFrame where to run tests defined in the scan definition.
+
+        Returns:
+            - A SodaSpark
```
Comment (desertaxle) [inline src/prefect/tasks/sodaspark/sodaspark_tasks.py:39]:
```suggestion
            - A SodaSpark ScanResult
```

File: tests/tasks/sodaspark/test_sodaspark_tasks.py [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+    def test_run_invalid_scan(self, monkeypatch):
+        scan_def = "invalid scan definition"
+        df = SparkSession.builder.getOrCreate().createDataFrame(
+            [{"id": 123, "value": "foo"}, {"id": 456, "value": "bar"}]
+        )
+        sb_mock = MagicMock()
+        sb_mock.side_effect = AttributeError
+        monkeypatch.setattr("prefect.tasks.sodaspark.sodaspark_tasks.scan", sb_mock)
+        soda_spark_scan_task = SodaSparkScan(scan_def=scan_def, df=df)
+        soda_spark_scan_task.run()
```
Comment (desertaxle) [inline tests/tasks/sodaspark/test_sodaspark_tasks.py:53]: What's the assertion for this test? Should the task raise an exception?

File: tests/tasks/sodaspark/test_sodaspark_tasks.py [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+        - min_length
+        tests:
+        - row_count > 0
+        """
+        df = "not a valid df"
+        sb_mock = MagicMock()
+        sb_mock.side_effect = AttributeError
+        monkeypatch.setattr("prefect.tasks.sodaspark.sodaspark_tasks.scan", sb_mock)
+        soda_spark_scan_task = SodaSparkScan(scan_def=scan_def, df=df)
+        soda_spark_scan_task.run()
```
Comment (desertaxle) [inline tests/tasks/sodaspark/test_sodaspark_tasks.py:70]: What's the assertion for this test? Should the task raise an exception?

**Review by desertaxle — CHANGES_REQUESTED** [review]
Looking good! I recommend adding some links to the Soda documentation to make it easier for users to understand what the inputs and outputs of this task are. Also, can you highlight some of the dependencies of this task are since it seems like it needs to run in a environment that has Java installed?

File: setup.py (hunk 2 above)
Comment (AlessandroLollo) [reply setup.py:93]: Exactly, `soda-spark` requires `soda-sql`.
I'm going to revert back the change, but I agree we should drop it in 1.0.

**Review by desertaxle — APPROVED** [review]

**Review by madkinsz — APPROVED** [review]

**Pull request merged**